Documentation/project-docs/cross-platform-cryptography.md

```diff
@@ -0,0 +1,241 @@
+
+ECDsaOpenSsl on macOS works if OpenSSL is installed in the system and an appropriate libcrypto dylib can be found via dynamic library loading, otherwise exceptions will be raised.
+
+### DSA
+
+DSA key generation is performed by the system libraries, and is subject to size limitations and performance characteristics thereof.
+
+| Function | Windows CNG | Linux | macOS | Windows CAPI |
+|----------|-------------|-------|-------|--------------|
+| Key creation (<= 1024 bits) | :white_check_mark: | :white_check_mark: | :x: | :white_check_mark: |
+| Key creation (> 1024 bits) | :white_check_mark: | :white_check_mark: | :x: | :x: |
+| Loading keys (<= 1024 bits) | :white_check_mark: | :white_check_mark:  | :white_check_mark: | :white_check_mark: |
+| Loading keys (> 1024 bits) | :white_check_mark: | :white_check_mark: | :question: | :x: |
+| FIPS 186-2 | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: |
+| FIPS 186-3 (SHA-2 signatures) | :white_check_mark: | :white_check_mark: | :x: | :x: |
+
+Windows CNG is used on Windows whenever `DSA.Create()` or `new DSACng()` is used, as well as the `.GetDSAPublicKey()` extension method for `X509Certificate2` and when possible for the `.GetDSAPrivateKey()` extension method for `X509Certificate2`.
+Windows CAPI is used on Windows whenever `new DSACryptoServiceProvider()` is used, as well as the `GetDSAPrivateKey()` extension method when the private key is not able to be opened with CNG (which requires a hardware-backed key with no CNG-capable driver).
+
+macOS seems to be capable of loading DSA keys whose size exceeds 1024-bit, but does not perform FIPS 186-3 behaviors with those keys, so the behavior of those keys is undefined.
+
+#### Native Interop
+
+.NET Core exposes types to allow programs to interoperate with the system libraries upon which the .NET cryptography code is layered.
+The types involved do not translate between platforms, and should only be directly used when necessary.
+
+| Type | Windows | Linux | macOS |
+|------|---------|-------|-------|
+| DSACryptoServiceProvider | :white_check_mark: | :question: | :question: |
+| DSACng | :white_check_mark: | :x: | :x: |
+| DSAOpenSsl | :x: | :white_check_mark: | :question: |
+
+DSAOpenSsl on macOS works if OpenSSL is installed in the system and an appropriate libcrypto dylib can be found via dynamic library loading, otherwise exceptions will be raised.
+
+On non-Windows systems RSACryptoServiceProvider can be used for compatibility with existing programs, but a `PlatformNotSupportedException` will be thrown from any method which requires system interop, such as opening a named key.
+
+## X.509 Certificates
+
+The majority of support for X.509 certificates in .NET Core comes from system libraries.
+All certificates are required to be loaded by the underlying system library to be loaded into an `X509Certificate2` instance in .NET Core (or an `X509Certificate` instance).
```

Comment:
Since certificates haven't really changed in the last 10 years, it means very little; but it means that .NET can't evolve faster than the system she's running on.